diff --git a/src/buffer.c b/src/buffer.c
index 227015ae..f3d8efaa 100644
--- a/src/buffer.c
+++ b/src/buffer.c
@@ -31,7 +31,9 @@ FILE_RCSID("@(#)$File: buffer.c,v 1.8 2020/02/16 15:52:49 christos Exp $")
 #endif	/* lint */
 
 #include "magic.h"
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <string.h>
 #include <stdlib.h>
 #include <sys/stat.h>
diff --git a/src/cdf.c b/src/cdf.c
index b5ad2f6a..e8bac9fe 100644
--- a/src/cdf.c
+++ b/src/cdf.c
@@ -43,7 +43,9 @@ FILE_RCSID("@(#)$File: cdf.c,v 1.120 2021/09/24 13:59:19 christos Exp $")
 #include <err.h>
 #endif
 #include <stdlib.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <string.h>
 #include <time.h>
 #include <ctype.h>
diff --git a/src/dprintf.c b/src/dprintf.c
index 027a64f3..873c5476 100644
--- a/src/dprintf.c
+++ b/src/dprintf.c
@@ -32,7 +32,9 @@ FILE_RCSID("@(#)$File: dprintf.c,v 1.2 2018/09/09 20:33:28 christos Exp $")
 #endif	/* lint */
 
 #include <assert.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdio.h>
 #include <stdarg.h>
 
diff --git a/src/file.c b/src/file.c
index 2f4f7a6d..0bd51d4f 100644
--- a/src/file.c
+++ b/src/file.c
@@ -38,7 +38,6 @@ FILE_RCSID("@(#)$File: file.c,v 1.190 2021/09/24 14:14:26 christos Exp $")
 #include "magic.h"
 
 #include <stdlib.h>
-#include <unistd.h>
 #include <string.h>
 #ifdef RESTORE_TIME
 # if (__COHERENT__ >= 0x420)
diff --git a/src/getline.c b/src/getline.c
index b00de01b..c2d41cae 100644
--- a/src/getline.c
+++ b/src/getline.c
@@ -33,7 +33,9 @@
 #if !HAVE_GETLINE
 #include <stdlib.h>
 #include <stdio.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <errno.h>
 #include <string.h>
 
diff --git a/src/is_csv.c b/src/is_csv.c
index 937ab5f4..2f704c51 100644
--- a/src/is_csv.c
+++ b/src/is_csv.c
@@ -166,7 +166,9 @@ file_is_csv(struct magic_set *ms, const struct buffer *b, int looks_text)
 #include <sys/stat.h>
 #include <stdio.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <stdint.h>
 #include <err.h>
diff --git a/src/is_json.c b/src/is_json.c
index 0b12438f..997a2767 100644
--- a/src/is_json.c
+++ b/src/is_json.c
@@ -436,7 +436,9 @@ file_is_json(struct magic_set *ms, const struct buffer *b)
 #include <sys/stat.h>
 #include <stdio.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <stdlib.h>
 #include <stdint.h>
 #include <err.h>
diff --git a/src/magic.c b/src/magic.c
index 03a456bc..d1810eb4 100644
--- a/src/magic.c
+++ b/src/magic.c
@@ -39,7 +39,9 @@ FILE_RCSID("@(#)$File: magic.c,v 1.115 2021/09/20 17:45:41 christos Exp $")
 #include "magic.h"
 
 #include <stdlib.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <string.h>
 #ifdef QUICK
 #include <sys/mman.h>
diff --git a/src/memtest.c b/src/memtest.c
index f9506f6e..ea502426 100644
--- a/src/memtest.c
+++ b/src/memtest.c
@@ -34,7 +34,9 @@
 #include <stdlib.h>
 #include <err.h>
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <dlfcn.h>
 #include <magic.h>
 
diff --git a/src/mygetopt.h b/src/mygetopt.h
index d7667626..b8e044c5 100644
--- a/src/mygetopt.h
+++ b/src/mygetopt.h
@@ -39,7 +39,9 @@
 #ifndef _GETOPT_H_
 #define _GETOPT_H_
 
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 
 /*
  * Gnu like getopt_long() and BSD4.4 getsubopt()/optreset extensions
diff --git a/src/pread.c b/src/pread.c
index 72d3a6b3..fecff913 100644
--- a/src/pread.c
+++ b/src/pread.c
@@ -3,7 +3,9 @@
 FILE_RCSID("@(#)$File: pread.c,v 1.3 2014/09/15 19:11:25 christos Exp $")
 #endif  /* lint */
 #include <fcntl.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 
 ssize_t
 pread(int fd, void *buf, size_t len, off_t off) {
diff --git a/src/readcdf.c b/src/readcdf.c
index 7622c7b0..977cb758 100644
--- a/src/readcdf.c
+++ b/src/readcdf.c
@@ -31,7 +31,9 @@ FILE_RCSID("@(#)$File: readcdf.c,v 1.74 2019/09/11 15:46:30 christos Exp $")
 
 #include <assert.h>
 #include <stdlib.h>
+#ifdef HAVE_UNISTD_H
 #include <unistd.h>
+#endif
 #include <string.h>
 #include <time.h>
 #include <ctype.h>
